### DISCUSSION TITLE
Using Machine-id with GCP Application

### DISCUSSION THREAD

**Discussion by doyelese** [gravitational/teleport#25597] [Q&A] [answered]
Hello, I am looking for some guidance on how to implement machine id to run GCP application commands.

I have used a combination of this guide to setup the GCP Application and have it working with my local user using the role: https://goteleport.com/docs/application-access/cloud-apis/google-cloud/

I have used this guide to get the Machine-id setup to use applications: 
https://goteleport.com/docs/machine-id/guides/applications/

The machine-id is running fine and authenticated with the application 'google-cloud-cli':

`teleport-ansible-service tbot[90698]: INFO [TBOT] Generated identity for app {"google-cloud-cli"} tbot/renew.go:446 `
`teleport-ansible-service tbot[90698]: INFO [TBOT] Persisted certificates successfully. Next renewal in approximately 20m0s. tbot/renew.go:753 `

I am not sure what commands would be used next to proxy my gcloud commands as the tutorial only provides curl examples where you would validate with some of the files found in /opt/machine-id.

My original thought would be that `tsh gcloud compute instance list` but that gives the error
```ERROR: current-profile is not set```

And i'm not certain how to setup the commands to use a tbot proxy after running a command like
`tbot proxy --destination-dir=/opt/machine-id --proxy=[teleport.helcim.com:443](http://teleport.helcim.com:443/) app --port=1234 google-cloud-cli`

I just need this authenticated machine to be able to run the gcloud compute instance list commands as the role it was setup with similar to how I run tsh gcloud commands after the tsh login.

Please assist! Thanks!

**Comment by webvictim** [comment, marked as answer]
Unforunately as per https://goteleport.com/docs/machine-id/introduction/ access to cloud provider APIs (AWS, GCP, Azure) using Machine ID is currently unsupported.

This feature request (https://github.com/gravitational/teleport/issues/16935) tracks the request for AWS Console access - when that is implemented, GCP and Azure shouldn't be far behind as they use similar underlying technology.

**Reply by doyelese** [reply]
Thank you for clarifying that `API Access` meant GCP CLI. I read that line in the documentation before but I'm not sure the wording was specific enough for me to understand it was referring to the CLI's included in the `Application` setup guides.
